mavros_extras/src/plugins/wheel_odometry.cpp

```diff
@@ -0,0 +1,525 @@
+			if (separation < 1.e-5f) {
+				odom_mode = OM::NONE;
+				ROS_WARN_NAMED("wo", "WO: Separation between the first two wheels is too small (%f).", separation);
+			}
+
+			// Check for reasonable radiuses
+			for (int i = 0; i < wheel_radius.size(); i++) {
+				if (wheel_radius[i] <= 1.e-5f) {
+					odom_mode = OM::NONE;
+					ROS_WARN_NAMED("wo", "WO: Wheel #%i has incorrect radius (%f).", i, wheel_radius[i]);
+				}
+			}
+		}
+
+		// Advertise RPM-s and distance-s
+		if (diagnostics) {
+			rpm_pub = wo_nh.advertise<mavros_msgs::Float32ArrayStamped>("rpm", 10);
+			dist_pub = wo_nh.advertise<mavros_msgs::Float32ArrayStamped>("distance", 10);
+		}
+
+		// Advertize topics
+		if (odom_mode != OM::NONE) {
+			if (twist_send)
+				twist_pub = wo_nh.advertise<geometry_msgs::TwistWithCovarianceStamped>("velocity", 10);
+			else
+				odom_pub = wo_nh.advertise<nav_msgs::Odometry>("odom", 10);
+		}
+		// No-odometry warning
+		else
+			ROS_WARN_NAMED("wo", "WO: No odometry computations will be performed.");
+
+	}
+
+	Subscriptions get_subscriptions()
+	{
+		return {
+			make_handler(&WheelOdometryPlugin::handle_rpm),
+			make_handler(&WheelOdometryPlugin::handle_wheel_distance)
+		};
+	}
+
+private:
+	ros::NodeHandle wo_nh;
+
+	ros::Publisher rpm_pub;
+	ros::Publisher dist_pub;
+	ros::Publisher odom_pub;
+	ros::Publisher twist_pub;
+	tf2_ros::TransformBroadcaster tfBroadcaster;
+
+	/// @brief Odometry computation modes
+	enum class OM {
+		NONE,	//!< no odometry computation
+		RPM,	//!< use wheel's RPM
+		DIST	//!< use wheel's cumulative distance
+	};
+	OM odom_mode; //!< odometry computation mode
+
+	int count;		//!< requested number of wheels to compute odometry
+	bool diagnostics;	//!< publish wheel's RPM and cumulative distance
+	std::vector<Eigen::Vector2f> wheel_offset; //!< wheel x,y offsets (m,NED)
+	std::vector<float> wheel_radius; //!< wheel radiuses (m)
+
+	bool twist_send;		//!< send geometry_msgs/TwistWithCovarianceStamped instead of nav_msgs/Odometry
+	bool tf_send;			//!< send TF
+	std::string frame_id;		//!< origin frame for topic headers
+	std::string child_frame_id;	//!< body-fixed frame for topic headers
+	std::string tf_frame_id;	//!< origin for TF
+	std::string tf_child_frame_id;	//!< frame for TF and Pose
+	float vel_cov;			//!< wheel velocity measurement error 1-var (m/s)
+
+	int count_meas;				//!< number of wheels in measurements
+	ros::Time time_prev;			//!< timestamp of previous measurement
+	std::vector<float> measurement_prev;	//!< previous measurement
+
+	bool yaw_initialized;			//!< initial yaw initialized (from IMU)
+
+	/// @brief Robot origin 2D-state (SI units)
+	Eigen::Vector3f rpose {0.f, 0.f, 0.f};		//!< pose (x, y, yaw)
+	Eigen::Vector3f rtwist {0.f, 0.f, 0.f};		//!< twist (vx, vy, vyaw)
+	Eigen::Vector3f rpose_cov {0.f, 0.f, 0.f};	//!< pose error 1-var (x_cov, y_cov, yaw_cov)
+	Eigen::Vector3f rtwist_cov {0.f, 0.f, 0.f};	//!< twist error 1-var (vx_cov, vy_cov, vyaw_cov)
```

Comment:
You may use `Eigen::Vector3f::Zero()` initializer.

Reply:
A bit, but actually class members initializers should be in constructor initializer list.